spec.md

```diff
@@ -2041,6 +2100,59 @@ Condition | gRPC Code | Description | Recovery Behavior
 | Call not implemented | 12 UNIMPLEMENTED | NodeGetInfo call is not implemented by the plugin or disabled in the Plugin's current mode of operation. | Caller MUST NOT retry. Caller MAY call `ControllerGetCapabilities` or `NodeGetCapabilities` to discover Plugin capabilities. |
 
 
+#### `NodeExpandVolume`
+
+A Node Plugin MUST implement this RPC call if it has `EXPAND_VOLUME` node capability.
+This RPC call allows CO to expand volume on the node.
+
+`NodeExpandVolume` MUST be called after `ControllerExpandVolume` if volume has `EXPAND_VOLUME` controller capability.
```

Comment:
Could you provide a concrete workflow in your mind about expanding a volume that is in use? It's still a bit unclear about how to use this call, and how to expand  in-use volumes in general.

Say we have a long-running service, e.g., mysql, that needs more storage. It would be easier to understand how these calls are used in a concrete workflow. For example, does the operator or the application (mysql) request the CO to issue these calls? Should the application be paused before requesting the volume to be expanded? It seems to me that we could understand the lifecycle requirements better if we can understand more about these questions.

Also it seems that this call is specifically designed for expanding in-use volumes, since it requires a volume path. Does that mean if the plugin does not support expanding in-use volumes, it should not have the `EXPAND_VOLUME` node capability?

Reply:
It is however true that, this API currently does not support expanding volumes that are not already "available" on a path on the node.  

To support something like that - `NodeExpandVolumeRequest` has to take same parameter as `NodeStageVolume`  - we can certainly do that.  But none of our use cases I have seen requires that. `NodeExpandVolume` is mostly designed to do file system expansion on the node and all modern file systems support expanding mounted volumes. Many file systems such as XFS even require that.

Having said that - If there is a future need, one can easily add to  `NodeExpandVolumeRequest` parameter and add support for such cases - I just do not see any use case for something like that.

Reply:
Hmm please forgive me for my poor use of words. Yeah by in-use volumes I meant volumes that are somehow "attached" to the node and thus have paths. I agree that there's no need to have this call if the target volume is not already available on the node, and was wondering that if we should change MAY to MUST to emphasize that this won't work on volumes that are not available to a node.

Reply:
Hmm I probably misunderstood what this "MAY" means. Say if the plugin has the `STAGE_UNSTAGE_VOLUME` capability, does that mean the CO can choose to call `NodeExpandVolume` either before or after `NodeStageVolume`?